hw2/03-count.js

```diff
@@ -1,3 +1,37 @@
 // Add your code here
 
-input.addEventListener('keydown', handleKeyDown);
+const highlightClass = "bg-warning";
+
+const textContainer = document.getElementById("text-container");
+// Restructure the original text container
+const originalTextSpanList = textContainer.innerText.split(" ").map((word) => {
+  const newSpan = document.createElement("span");
+  newSpan.innerText = word;
+  newSpan.style.transitionDuration = "100ms";
+  return newSpan;
+});
+
+textContainer.innerHTML = "";
+originalTextSpanList.forEach((wordSpan) => {
+  textContainer.appendChild(wordSpan);
+  textContainer.appendChild(document.createTextNode(" "));
+});
+
+const handleHighlight = (textToHighlight) => {
+  originalTextSpanList.forEach((wordSpan) => {
+    const word = wordSpan.innerText;
+    const textToHighlightList = textToHighlight.split(" ");
+
+    wordSpan.classList[textToHighlightList.includes(word) ? "add" : "remove"](
+      highlightClass,
+    );
+  });
+};
+
+const input = document.getElementById("input");
+const handleKeyDown = (event) => {
+  const text = event.target.value;
+  handleHighlight(text);
+};
+
+input.addEventListener("input", handleKeyDown);
```

Comment:
I think handling Key events instead of input is suggested.

Reply:
hmmm, `keydown` captures the input state at the moment of key down (any button press when the `input` is focused). However, the change to the actual input value comes after. Therefore, the `event.target.value` will have the value of the previous state of the input of the latest state

I made this [playground](https://codepen.io/tunk/pen/nrXmNJ) in case you want to double-check 🥪

Reply:
Thank you for explaining the difference. I went further and researched about it and found [this](https://stackoverflow.com/questions/38502560/whats-the-difference-between-keyup-keydown-keypress-and-input-events).
I checked the playground code, but it is handling the `submit` event. Is it the right link?

Reply:
oh oops sorry I forgot to save 

here you go https://codepen.io/ptmdmusique/pen/jOeLWMW